ee/startup/src/crt0.s

```diff
@@ -174,3 +174,9 @@ _root:
    .align   6
 _args:
    .space   4+16*4+256   # argc, 16 arguments, 256 bytes payload
+
+   .sdata
+   .global __dso_handle
+   .weak __dso_handle
+__dso_handle:
+   .long 0
```

Comment:
crt0.o should actually be defined in newlib itself.
Is this for c++ destructors?

Reply:
Probably. I was getting linking error, so I decided to search newlib code for the symbol. It was defined that way so I copied it.